src/mothball.go

```diff
@@ -143,12 +145,32 @@ func (m *Mothball) Refresh() error {
 	if err != nil {
 		return err
 	}
-
+	
 	if m.zf != nil {
 		m.zf.Close()
 	}
 	m.zf = zf
 	m.mtime = mtime
+	
+	os.RemoveAll(path.Join(m.filename[:len(m.filename)-3]))
+	os.Mkdir(path.Join(m.filename[:len(m.filename)-3]), 0755)
```

Comment:
This filename manipulation and directory creation makes me nervous.

1. What if m.filename doesn't end in ".mb" for some reason?
2. What do we do when someone make huge changes to a mothball and there are a bunch of files dangling around? We need a way to clean things up, which means we need to remember what we created...

Could we instead just pull the graders out? I guess not, since they might pull in other files. I guess the problem is that we're mixing data and executables where before we only had data. This is making me have second thoughts about the executable-in-mothball thing.

Reply:
I can make the filename filter a little smarter, just doing a linear search for ".".  That said, I think functionally this is still actually OK, it would just name the directory something like "afile.jp" for a mothball named "afile.jpeg".

I am also hesitant on the extracting of the mothballs.  We could do just the grader if we wanted to, all that is guaranteed for the grader are the specified files so we would just pull those out too.  But still, we are extracting things to the filesystem which I think is a little uglier (and less efficient space wise, albeit more efficient computation wise) than leaving the files as mothballs.

Reply:
As I was going through the code yesterday making some fixes to the fork, I noticed that we actually require the ".mb" extension for mothballs elsewhere in the code, which would make this code pretty correct.  We could change that though, allowing mothballs to use any extension.